src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,92 @@
+import { getUser } from '../../api';
+import { useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = currentTodo as Todo;
+
+  const loadUser = async () => {
+    setLoading(true);
+
+    try {
+      const loadedUser = await getUser(userId);
+
+      setUser(loadedUser);
+      setLoading(false);
+    } catch {
+      throw new Error('error');
+    }
+  };
+
+  useEffect(() => {
+    loadUser();
+  }, []);
+
+  const deleteSelectedTodo = () => {
+    dispatch(currentTodoActions.removeTodo());
+  };
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {loading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              {`Todo #${id}`}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+            <button
+              type="button"
+              aria-label="Mute volume"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => deleteSelectedTodo()}
+            />
+          </header>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {title}
+            </p>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">fugiat veniam minus</p>
+            <p className="block" data-cy="modal-user">
+              {completed
+                ? (<strong className="has-text-success">Done</strong>)
+                : (<strong className="has-text-danger">Planned</strong>)}
+              {' by '}
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+              {user && (
+                <a href={`mailto:${user?.email}`}>
+                  {user?.name}
+                </a>
+              )}
```

Comment:
You have a check if the user exists, so `?` are redundant here 
```suggestion
              {user && (
                <a href={`mailto:${user.email}`}>
                  {user.name}
                </a>
              )}
```